cpp/tests/worker.cpp

```diff
@@ -43,16 +56,18 @@ class WorkerCapabilityTest : public ::testing::Test,
   }
 };
 
-class WorkerProgressTest : public WorkerTest,
-                           public ::testing::WithParamInterface<std::tuple<bool, ProgressMode>> {
+class WorkerProgressTest
+  : public WorkerTest,
+    public ::testing::WithParamInterface<std::tuple<bool, ProgressMode, ExtraParams>> {
```

Comment:
There's no way for derived classes to inherit again from `WithParamInterface`, thus added an `ExtraParams` attribute to the base class that can be used by derived classes to signify other things to be tested. Not completely happy with this solution, but the alternative seems to be duplicating the entire `WorkerProgressTest` class, potentially more than once.